src/os/bluestore/bluestore_types.h

```diff
@@ -156,7 +156,8 @@ struct denc_traits<vector<bluestore_pextent_t>> {
   static void bound_encode(const vector<bluestore_pextent_t>& v, size_t& p) {
     p += sizeof(uint32_t);
     size_t per = 0;
-    denc(*(bluestore_pextent_t*)nullptr, per);
+    bluestore_pextent_t tmp;
```

Comment:
When I asked @liewegas, he said this use of the null pointer is used as a debugging aid, to make sure that we only call functions that properly implement bounded encoding from functions that implement bound encoding. (That is, ones that use only sizeof, I think.)

The easiest way to fix this problem would probably be to either use static_assert, or replace the bound_encoding trait with a static_bound trait. (This would mean that we can't naively use bounded encding with features. I'm not sure if this is actually bad in practice. We could, if we wanted, set things up to return a size at compile-time assuming the features are known, which might not be bad in general since the set of features varying any given object will be fairly small, I expect.)